### ISSUE TITLE
solution generated for VS2012 (and possibly VS2010) has a slight issue

### ISSUE THREAD

**Issue by dgregorycryptic** [premake/premake-core#121]
If I build a standard premake5.lua with "premake5 vs2012", and have a "dependson" directive in one of the projects, an old-style solution dependency is still emitted.  The project references you have set up inside of .vcxproj files is sufficient.

The specific code that is generating this is the "premake.vstudio.sln2005.projects" function.  The code it calls (sln2005.projectdependencies) needs to be skipped for vs2012 (and possibly vs2010).

**Comment by tvandijck** [comment]
we got a fix for that here:
https://github.com/Blizzard/premake-core/commit/dd9e33e34563bc5849eb33e030c998153c42000d

I can make a pull request for it if people are interested.

**Comment by tvandijck** [comment]
FWIW, this fixes the issue more complicated then what is suggested above... that is because if you use a 'dependson' you don't want it to show up in the reference list, since a dependson is not supposed to be a link dependency, but just a build order dependency...

So the fixes we made emit references for links, and project dependencies for dependson.

**Comment by dgregorycryptic** [comment]
Should "links" (for a project or externalproject) indicate a build order dependency also?

**Comment by tvandijck** [comment]
I don't think so, whenever it is added in the 'references' it will automatically be a build order dependency...

if you manually authored a project in visual studio with a bunch of references, you end up with a solution that has no dependencies setup in the SLN at all... just references in the vcxproj... only when you then explicitly set build order dependencies do you get things added to the SLN... but you'll notice that if you have a reference to a project, the build order dependency is already checked.
